core/src/storage/collections/btree_map/node.rs

```diff
@@ -0,0 +1,439 @@
+        &self.key
+    }
+
+    /// Returns the `value` stored in this key/value pair.
+    pub(super) fn value(self) -> V {
+        self.value
+    }
+
+    /// Returns a mutable reference to the `value` stored in this key/value pair.
+    pub(super) fn value_ref_mut(&mut self) -> &mut V {
+        &mut self.value
+    }
+
+    /// Returns a reference to the `value` stored in this key/value pair.
+    pub(super) fn value_ref(&self) -> &V {
+        &self.value
+    }
+}
+
+impl<K, V> Flush for KVPair<K, V>
+where
+    K: Encode + Flush,
+    V: Encode + Flush,
+{
+    #[inline]
+    fn flush(&mut self) {
+        self.key.flush();
+        self.value.flush();
+    }
+}
+
+/// Reference to a key/value pair in the tree.
+pub(super) struct KVRef<'a, K, V> {
+    /// Reference to the key.
+    key: &'a K,
+    /// Reference to the value.
+    value: &'a V,
+}
+
+impl<'a, K, V> KVRef<'a, K, V> {
+    /// Creates a `KVRef` from a `KVPair`.
+    pub(super) fn new(pair: &'a KVPair<K, V>) -> Self {
+        Self {
+            key: pair.key_ref(),
+            value: pair.value_ref(),
+        }
+    }
+
+    /// Returns a tuple with references to the `key` and `value`
+    /// stored in this key/value pair..
+    pub(super) fn kv(self) -> (&'a K, &'a V) {
+        (self.key, self.value)
+    }
+
+    /// Returns a reference to the `key` stored in this key/value pair.
+    pub(super) fn key(self) -> &'a K {
+        self.key
+    }
+}
+
+/// A node in the tree.
+///
+/// Each node contains `CAPACITY` keys and values and an edges array over
+/// which children nodes can be linked. Each child has a link to its parent.
+///
+/// Each node is stored as one storage entity. This reduces storage access,
+/// since with each fetch the entire content of a node (all its elements, etc.)
+/// are fetched.
+#[derive(PartialEq, Eq, Encode, Decode)]
+#[cfg_attr(feature = "ink-generate-abi", derive(Metadata))]
+pub(super) struct Node<K, V> {
+    /// A reference to this node's parent node.
+    parent: Option<NodeHandle>,
+
+    /// This node's index into the parent node's `edges` array.
+    /// If, for example, `parent_idx = Some(2)` this refers to the
+    /// second position in the `edges` array of its parent node.
+    parent_idx: Option<u32>,
+
+    /// The array storing a pointer to the key/value pairs in a node.
+    /// For performance reasons each node contains only a pointer to the index where
+    /// the key/value pair is stored in `kv_pairs`. Otherwise re-balancing the tree
+    /// or moving entries would be very expensive. With this indirection we only
+    /// have to move the `u32` index.
+    pub(super) pairs: [Option<KVStoragePointer>; CAPACITY],
+
+    /// The pointers to the children of this node.
+    pub(super) edges: [Option<NodeHandle>; EDGES],
+
+    /// Number of elements stored in this node.
+    len: u32,
+
+    /// Marker for compile-time checking of correct key/value pair types.
+    marker: PhantomData<(K, V)>,
```

Comment:
```suggestion
    marker: PhantomData<fn() -> (K, V)>,
```

Reply:
Where do you see the advantage of the `fn() -> ` here?